tests/app/questionnaire/test_placeholder_transforms.py

```diff
@@ -262,6 +262,149 @@ def test_telephone_number_link(self):
             == '<a href="tel:01234567890">012345 67890</a>'
         )
 
+    def date_range_bounds(self, params):
+        with self.subTest():
+            for i, (
+                ref_date,
+                weeks_prior,
+                day_range,
+                first_day_of_week,
+                (date_from, date_to),
+            ) in enumerate(params):
+                expected = (date_from, date_to)
+                actual = self.transforms.date_range_bounds(
+                    ref_date, weeks_prior, day_range, first_day_of_week
+                )
+                self.assertEqual(expected, actual)
+
+    def test_date_range_bounds_all_weekdays(self):
+        params = [
+            ("2021-09-26", -1, 7, "MONDAY", ("2021-09-13", "2021-09-19")),
+            ("2021-09-27", -1, 7, "MONDAY", ("2021-09-20", "2021-09-26")),
+            ("2021-09-28", -1, 7, "MONDAY", ("2021-09-20", "2021-09-26")),
+            ("2021-09-29", -1, 7, "MONDAY", ("2021-09-20", "2021-09-26")),
+            ("2021-09-30", -1, 7, "MONDAY", ("2021-09-20", "2021-09-26")),
+            ("2021-10-01", -1, 7, "MONDAY", ("2021-09-20", "2021-09-26")),
+            ("2021-10-02", -1, 7, "MONDAY", ("2021-09-20", "2021-09-26")),
+            ("2021-10-03", -1, 7, "MONDAY", ("2021-09-20", "2021-09-26")),
+            ("2021-10-04", -1, 7, "MONDAY", ("2021-09-27", "2021-10-03")),
+        ]
+
+        self.date_range_bounds(params)
```

Comment:
Thought of renaming `self.date_range_bounds(params)` to like `self.assert_date_range_bounds(params)`, just to give it some context on what it is doing.

My concern with these tests is that it seems a mix of patterns from pytest/subtest, i.e it isn't clear what the params are until you check the `date_range_bounds`, but that wouldn't be the case with pytest paramaterize. We should good to move away and just use pytest. Keep it as is for now I guess. I was going to suggest maybe a comment on what each param was but would need to duplicate for each test.

Reply:
For context for other, the new test have been moved to `pytest` for clarity. We can migrate existing one as a separate piece.